### PR TITLE
#542: Change countries to be separated by commas on detail page

### PR DESCRIPTION
**Pull request by markusjt**
Decided to create a new method for joining the values (even if the one-liner is simple) to make it easier to check the actual result in a test.

### CODE REVIEW

File: src/components/Detail.tsx [hunk 1]
```diff
@@ -104,6 +104,21 @@ export default class Detail extends React.Component<Props, State> {
...
+   * as long as the values are strings and truthy.
+   *
+   * @param arr array of objects
+   * @param field name of the field to look for values
+   * @param char character to use for joining (space added by default)
+   * @returns <div> element that contains values separated by a character
+   */
+  static joinFieldValuesByCharacter(arr: Array<{[key:string]: any}>, field: string, char: string) {
+    return (
+      <div>{arr.filter(obj =>  obj[field] && typeof obj[field] === 'string').map(obj => obj[field]).join(`${char} `)}</div>
```
Comment (matthew-morris-cessda) [inline src/components/Detail.tsx:118]: `arr.filter(obj =>  obj[field] && typeof obj[field] === 'string').map(obj => obj[field])` is needlessly complex when both the object and a function to extract the value from the object could be provided, e.g.

```javascript
static joinFieldValuesByCharacter<T>(arr: Array<T>, extractor: (object: T) => string, separator: string) {
  return (
    <div>{arr.map(element => extractor(element)).join(separator)}</div>
  )
}
```

The calling code:

```javascript
Detail.joinFieldValuesByCharacter(item.studyAreaCountries, c => c.country, ", ")
```

This also has the advantage of being type safe as well as being more flexible if the extractor is non-trivial.

The final `join()` inserts a space regardless of the provided separator, which may be surprising.

File: tests/src/components/Detail.tsx
Comment (matthew-morris-cessda) [inline tests/src/components/Detail.tsx:1]: Given that `Country.country` is guaranteed to be a string in the type definition ([see the type's definition](https://github.com/cessda/cessda.cdc.searchkit/blob/5f955b3f691be1bca6a90bf69f28ceaf8314dc0e/common/metadata.ts#L79)), there is no need to test the `null`, `undefined` and `notcountry` cases.

**Review by matthew-morris-cessda — CHANGES_REQUESTED** [review]
I have reviewed the code, and have some remarks.

File: src/components/Detail.tsx (hunk 1 above)
Comment (markusjt) [reply src/components/Detail.tsx]: Thank you for this! I made these changes but I kept filter as a part of it to deal with "" (and " ") just in case.

**Review by matthew-morris-cessda — APPROVED** [review]
